src/expr/mod.rs

```diff
@@ -28,7 +28,9 @@ use std::{
 };
 
 use crate::{
-    data_type::{self, value, DataType, DataTyped, Variant as _},
+    data_type::{
+        self, function::Function as _, value, DataType, DataTyped, Optional, Variant as _,
```

Comment:
Why this? 
`function::Function as _`

Reply:
We need "function::Function as _" for using the super_intersection but 'Optional' should not be there